models/user.model.js

```diff
@@ -1,100 +1,81 @@
-const mongoose = require('mongoose');
+const mongoose = require("mongoose");
 const Schema = mongoose.Schema;
-const getBaseUrl = require('../middleware/getBaseUrl')
+const getBaseUrl = require("../middleware/getBaseUrl");
 
-const passport = require('passport')
-const passportLocalMongoose = require("passport-local-mongoose");
+const passport = require("passport");
+//const passportLocalMongoose = require("passport-local-mongoose");
 const GoogleStrategy = require("passport-google-oauth20").Strategy;
-const findOrCreate = require('mongoose-findorcreate')
+//const findOrCreate = require("mongoose-findorcreate");
 
 const UserSchema = new Schema({
-    username: String,
-    firstName: String,
-    lastName: String,
-    email: String,
-    googleId: String,
-    secret: String,
-    imgUrl: String,
-    bio: {
+    id: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true,
+    },
+    username: {
+        type: String,
+        required: true,
+    },
+    firstName: {
         type: String,
-        default: ""
+        required: true,
     },
-    major: {
+    lastName: {
         type: String,
-        default: "N/A"
+        required: true,
     },
-    schedule: {
-        type: [{
-            _id: String,
-            department: String,
-            number: Number,
-            professor: String,
-            startTime: String,
-            endTime: String,
-            monday: Boolean,
-            tuesday: Boolean,
-            wednesday: Boolean,
-            thursday: Boolean,
-            friday: Boolean,
-            saturday: Boolean,
-            sunday: Boolean,
-            lab: Boolean,
-            firstHalfMod: Boolean,
-            secondHalfMod: Boolean
-        }],
-        default: []
+    email: {
+        type: String,
+        required: true,
     },
-    hideSchedule: {
-        type: Boolean,
-        default: false
+    password: {
+        type: String,
+        required: true,
     },
-    dateReg: {
-        type: Date,
-        default: Date.now
-    }
+    googleId: {
+        type: String,
+        required: false,
+    },
+    secret: String,
+    imgUrl: String,
+    phone: String,
 });
 
-UserSchema.plugin(passportLocalMongoose);
-UserSchema.plugin(findOrCreate);
-
-module.exports = User = mongoose.model('user', UserSchema);
+module.exports = User = mongoose.model("user", UserSchema);
 
-passport.use(User.createStrategy());
```

Comment:
Is there a reason this section below was deleted ? I forget whether it is necessary or not.

Reply:
I was throwing an error on it - let me give it another shot